app/scripts/controllers/dashboard.coffee

```diff
@@ -88,10 +88,15 @@ angular.module('edudashAppCtrl').controller 'DashboardCtrl', [
         watchCompute = watchComputeSrv $scope
 
         watchCompute 'metric',
-          dependencies: ['schoolType', 'rankBy']
-          computer: ([schoolType, criteria]) ->
+          dependencies: ['schoolType', 'rankBy', 'visMode']
+          computer: ([schoolType, criteria, visMode], [oldSchoolType, oldCriteria, oldVisMode]) ->
             unless schoolType? and criteria?
               null
+            else if oldVisMode? and  visMode != oldVisMode
```

Comment:
Hmm, yeah maybe. The logic is so complex you know :stuck_out_tongue: 

the `visMode != oldVisMode` seems suspicious to me. Maybe we can just change `brackets.getMetric` to (optionally?) take `visMode` as a param?